File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/model/Topping.java
package dam.isi.frsf.utn.edu.ar.delivery.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

import dam.isi.frsf.utn.edu.ar.delivery.constants.appConstants;

public class Topping implements Serializable, Addin {

    private final static long serialVersionUID = 1588113798071471128L;
    @SerializedName("label")
    @Expose
    private String label;
    @SerializedName("imgURL")
    @Expose
    private String imgURL;

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public Topping withLabel(String label) {
        this.label = label;
        return this;
    }

    public String getImgURL() {
        return imgURL;
    }

    public void setImgURL(String imgURL) {
        this.imgURL = imgURL;
    }

    public Topping withImgURL(String imgURL) {
        this.imgURL = imgURL;
        return this;
    }

    public String getCompleteImgURL() {
        return appConstants.staticPath + "toppingImages/" + imgURL;
    }

    @Override
    public String toString() {
        return label;
    }
}

File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/model/Topping.kt
package dam.isi.frsf.utn.edu.ar.delivery.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import dam.isi.frsf.utn.edu.ar.delivery.constants.appConstants

data class Topping(
        @SerializedName("label")
        @Expose
        override var label: String = "",
        @SerializedName("imgURL")
        @Expose
        var imgURL: String = ""
) : Addin, Parcelable {
    private val checked: Boolean? = null

    fun withLabel(label: String): Topping {
        this.label = label
        return this
    }

    fun withImgURL(imgURL: String): Topping {
        this.imgURL = imgURL
        return this
    }

    override val completeImgURL: String
        get() = appConstants.staticPath + "toppingImages/" + imgURL

    override fun toString(): String {
        return label
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Topping> = object : Parcelable.Creator<Topping> {
            override fun createFromParcel(source: Parcel): Topping = Topping(source)
            override fun newArray(size: Int): Array<Topping?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(label)
        dest.writeString(imgURL)
    }
}